rskj-core/src/main/java/co/rsk/db/migration/OrchidToUnitrieMigrator.java

```diff
@@ -116,10 +118,17 @@ public static void migrateStateToUnitrieIfNeeded(RskContext ctx) throws IOExcept
             return;
         }
 
+        BlockStore blockStore = ctx.getBlockStore();
+        long maximumBlockNumberToMigrate = ctx.getRskSystemProperties().getDatabaseMigrationMaximumHeight();
+        if (maximumBlockNumberToMigrate != 0 && blockStore.getMaxNumber() > maximumBlockNumberToMigrate) {
+            ((IndexedBlockStore) blockStore).rewind(maximumBlockNumberToMigrate);
```

Comment:
If we are keeping this class I would either receive an indexedblockstore or add the rewind functionality to the interface.

Reply:
The general idea is to remove this whole class in the next or the next+1 version, with that in mind we are not sure if this will be the final interface being used for when we implement the _rewind_ feature, that's why this wasn't promoted to the interface.